File: articles.jdbc/src/main/java/pet/db/jdbc/repository/ReviewRepository.java
package pet.db.jdbc.repository;

import jakarta.validation.constraints.NotNull;

import pet.db.jdbc.model.dto.Review;

import java.util.List;
import java.util.Optional;

public interface ReviewRepository {

    Review save(@NotNull Review review);

    void deleteById(@NotNull Integer id);

    Optional<Review> findById(@NotNull Integer id);

    List<Review> findByAuthorId(@NotNull Integer authorId);

    List<Review> findByArticleId(@NotNull Integer articleId);

}

File: src/main/kotlin/pet/articles/repository/review/ReviewRepository.kt
package pet.articles.repository.review

import pet.articles.model.dto.Review
import pet.articles.repository.CrudRepository

interface ReviewRepository : CrudRepository<Review> {

    fun findByAuthorId(authorId: Int): List<Review>

    fun findByArticleId(articleId: Int): List<Review>
}